portfolio/src/main/webapp/task.js

```diff
@@ -6,20 +11,164 @@
  * converted into a JSON string.
  */
 class Task {
-  // TODO(raulcruise): could you add the type for each param please? 
-  
+  // TODO(raulcruise): could you add the type for each param please?
+
   /**
    * All fields are required except description, which will be null
    * when not set.
    * @param name: name of task as String
    * @param description: description of task as String, can be null
-   * @param duration: duration of task as Duration class
-   * @param priority: priority level of task, range 1-5 inclusive. 
+   * @param duration: duration of task in minutes as an integer
+   * @param taskPriority: priority level of task, range 1-5 inclusive.
    */
-  constructor(name, description, duration, priority) {
+  constructor(name, description, duration, taskPriority) {
     this.name = name;
     this.description = description;
     this.duration = duration;
-    this.priority = priority;
+    this.taskPriority = taskPriority;
+  }
+}
+
+/**
+ * This method grabs the required inputs from the front end of the page that
+ * will populate the Task's information. It requires that the front end fields
+ * for name, and duration to be passed in. Once it checks for the required
+ * inputs, a JavaScript task class is created and passed into the
+ * updateTaskList function. Return statements are used only to end the function.
+ */
+function createNewTask() {
+  const name = $('#new-task-name').val();
+  const description = $('#new-task-description').val();
+  const length = parseInt($('#new-task-estimated-length').val());
+  const lengthUnit = $('#new-task-estimated-length-unit').val();
+  const priority = parseInt($('#new-task-priority').val());
+
+  if (validateTaskName(name).isValid) {
+    $('#empty-name-message').hide();
```

Comment:
It might be better to consistently use the d-none class name.

ex:
$('#...').addClass('d-none');

Here and below.

Reply:
Done.